File: reactor-netty2/src/main/java/com/example/demo/HelloController.java
package com.example.demo;


import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/greeting")
public class HelloController {

    @GetMapping
    public Mono<String> sayHello(@RequestParam("name") String name){
        return Mono.just("Hello, "+ name);
    }

}

File: kotlin-coroutines/src/main/kotlin/com/example/demo/HelloController.kt
package com.example.demo

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
@RequestMapping("/hello")
class HelloController {

    @GetMapping
    fun hello(): String {
        return "Hello Kotlin Coroutines at" + LocalDateTime.now()
    }
}